pysmt/smtlib/solver.py

```diff
@@ -22,24 +22,61 @@
 from pysmt.solvers.eager import EagerModel
 from pysmt.smtlib.parser import SmtLibParser
 from pysmt.smtlib.script import SmtLibCommand
-from pysmt.solvers.solver import Solver
+from pysmt.solvers.solver import Solver, SolverOptions
 from pysmt.exceptions import (SolverReturnedUnknownResultError,
                               UnknownSolverAnswerError)
 
+
+class SmtLibOptions(SolverOptions):
+    """Options for the SmtLib Solver.
+
+    * debug_interaction: True, False
+      Print the communication between pySMT and the wrapped executable
+    """
+    def __init__(self, **base_options):
+        SolverOptions.__init__(self, **base_options)
+        if self.unsat_cores_mode is not None:
+            raise ValueError("'unsat_cores_mode' option not supported.")
+        self.debug_interaction = False
+
+        if 'debug_interaction' in self.solver_options:
+            self.debug_interaction = self.solver_options
+
+    def __call__(self, solver):
+        # These options are needed for the wrapper to work
+        solver.set_option(":print-success", "true")
+        solver.set_option(":diagnostic-output-channel", '"stdout"')
+
+        if self.generate_models:
+            solver.set_option(":produce-models", "true")
+        else:
+            solver.set_option(":produce-models", "false")
+
+        if self.random_seed is not None:
+            solver.set_option(":random-seed", str(self.random_seed))
+
+        for k,v in self.solver_options.items():
+            if k in (':print-success', 'diagnostic-output-channel'):
```

Comment:
What if the user pusts ```:produce-models``` in the solver_options? Maybe we should avoid overriding all the pysmt-interpreted options.